esphome/components/logger/__init__.py

```diff
@@ -171,8 +171,8 @@ def validate_local_no_higher_than_global(value):
                 CONF_HARDWARE_UART,
                 esp8266=UART0,
                 esp32=UART0,
-                esp32_s2=USB_CDC,
-                esp32_s3=USB_CDC,
+                esp32_s2=USB_SERIAL_JTAG,
+                esp32_s3=USB_SERIAL_JTAG,
```

Comment:
In light of #5928, we can (and should) do this now:
```suggestion
                esp32_s2=USB_CDC,
                esp32_s3=USB_SERIAL_JTAG,
                esp32_s3_arduino=USB_CDC,
```
...the S2 does not support JTAG at all and CDC with the S3 won't work correctly with IDF.